baseball/src/Interface/Interface.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState, useEffect, useReducer } from "react";
+import Display from "../Display/Display";
+import Dashboard from "../Dashboard/Dashboard";
+
+import InterfaceDispatch from "./InterfaceDispatch";
+
+function Interface() {
+  const [state, dispatch] = useReducer(
+    (state, action) => {
+      switch (action.type) {
+        case "increment_balls":
+          if (state.balls + 1 > 3) return { ...state, balls: 0, strikes: 0 };
+          return { ...state, balls: state.balls + 1 };
+        case "increment_strikes":
+          if (state.strikes + 1 > 2) return { ...state, balls: 0, strikes: 0 };
+          return { ...state, strikes: state.strikes + 1 };
+        case "record_foul":
+          if (state.strikes + 1 > 2) return { ...state, strikes: 2 };
+          return { ...state, strikes: state.strikes + 1 };
+        case "record_hit":
+          return { ...state, strikes: 0, balls: 0 };
+      }
+    },
+    {
+      balls: 0,
+      strikes: 0
+    }
```

Comment:
This same object seems to be passed around a lot; you could always make a variable for it:
```javascript
const defaultState = {
  balls: 0,
  strikes: 0
};

const [state, dispatch] = useReducer(
  (state, action) => {
    switch (action.type) {
      case "increment_balls":
        if (state.balls + 1 > 3) return { ...state, ...defaultState };
        return { ...state, balls: state.balls + 1 };
      case "increment_strikes":
        if (state.strikes + 1 > 2) return { ...state, ...defaultState };
        return { ...state, strikes: state.strikes + 1 };
      case "record_foul":
        if (state.strikes + 1 > 2) return { ...state, strikes: 2 };
        return { ...state, strikes: state.strikes + 1 };
      case "record_hit":
        return { ...state, ...defaultState };
    }
  },
  { ...defaultState }
);
```
I'd also consider using some destructuring for state as well so as to reduce repetition:
```javascript
const defaultState = {
  balls: 0,
  strikes: 0
};

const [state, dispatch] = useReducer(
  (state, action) => {
    const { balls, strikes } = state;
    switch (action.type) {
      case "increment_balls":
        if (balls + 1 > 3) return { ...state, ...defaultState };
        return { ...state, balls: balls + 1 };
      case "increment_strikes":
        if (strikes + 1 > 2) return { ...state, ...defaultState };
        return { ...state, strikes: strikes + 1 };
      case "record_foul":
        if (strikes + 1 > 2) return { ...state, strikes: 2 };
        return { ...state, strikes: strikes + 1 };
      case "record_hit":
        return { ...state, ...defaultState };
    }
  },
  { ...defaultState }
);
```
Furthermore, I'd try to check values without extra reading (with the addition you're currently using) for people like me who happen to be looking at your code:
```javascript
const defaultState = {
  balls: 0,
  strikes: 0
};

const [state, dispatch] = useReducer(
  (state, action) => {
    const { balls, strikes } = state;
    switch (action.type) {
      case "increment_balls":
        if (balls > 2) return { ...state, ...defaultState };
        return { ...state, balls: ++balls };
      case "increment_strikes":
        if (strikes > 1) return { ...state, ...defaultState };
        return { ...state, strikes: ++strikes };
      case "record_foul":
        if (strikes > 1) return { ...state, strikes: 2 };
        return { ...state, strikes: ++strikes };
      case "record_hit":
        return { ...state, ...defaultState };
    }
  },
  { ...defaultState }
);
```
(I also opted for pre-increment operators since we're no longer accessing the state value but just a copy instead.)

Reply:
One other thing I've seen when people leave out braces in their `if` statements that have a return with an effective `else` in turn is the use of that `else` still (purely for readability's sake):
```javascript
const [state, dispatch] = useReducer(
  (state, action) => {
    const { balls, strikes } = state;
    switch (action.type) {
      case "increment_balls":
        if (balls > 2) return { ...state, ...defaultState };
        else return { ...state, balls: ++balls };
      case "increment_strikes":
        if (strikes > 1) return { ...state, ...defaultState };
        else return { ...state, strikes: ++strikes };
      case "record_foul":
        if (strikes > 1) return { ...state, strikes: 2 };
        else return { ...state, strikes: ++strikes };
      case "record_hit":
        return { ...state, ...defaultState };
    }
  },
  { ...defaultState }
);
```